django/db/models/sql/query.py

```diff
@@ -1202,6 +1191,89 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
     def add_filter(self, filter_clause):
         self.add_q(Q(**{filter_clause[0]: filter_clause[1]}))
 
+    def find_base_splits(self, q_object, found_splits, in_negated=False, path=None):
```

Comment:
This method is quite confusing to me. It's called recursively, but doesn't return anything. Is it just mutating the state passed in? Anyway to make this clearer?